tests/test_api_routes.py

```diff
@@ -1,9 +1,22 @@
 from unittest.mock import patch, Mock
 import requests
 import pytest
+from stampy_nlp.settings import ALLOWED_ORIGINS
 from data.duplicates import STAMPY_DUPLICATES
 
 
+@pytest.mark.parametrize('origin', ALLOWED_ORIGINS)
+def test_cors(client, origin):
+    response = client.get('/api/log_query', headers={'Origin': origin})
+    assert response.headers.get('Access-Control-Allow-Origin') == origin
+
+
+@pytest.mark.skipif(ALLOWED_ORIGINS == '*', reason='CORS is not being used')
```

Comment:
I don't like this magic. This causes the tests to be sort of non deterministic, as they depend on environment variables. That being said, I like the idea of dynamically checking all whitelisted domains, so dunno...